lib/index.ts

```diff
@@ -85,8 +96,16 @@ export class SyncClient<T=any> {
 
   public async writeMessage(message: any) {
     if (this.state !== "awaitingMessage") {
-      throw new Error("Not waiting for message");
+      if (this._awaitingMessageResolve) {
+        throw new Error("Not waiting for message, and another write is already queued.")
+      }
+
+      await new Promise<void>((resolve) => {
+        this._awaitingMessageResolve = resolve;
+      });
+      delete this._awaitingMessageResolve;
```

Comment:
I'm a bit confused by this part. You save the resolve callback but then immediately delete it. How likely is it that this value is defined when used in call? Why delete it here already?

Reply:
Since the Promise is awaited, it should be impossible to reach the delete line before the resolve callback is called in `call`.

It's deleted so that the next time writeMessage is called early it won't fail at the check a couple of lines up.